rs/tests/driver/src/driver/test_env_api.rs

```diff
@@ -1742,18 +1751,29 @@ pub trait HasPublicApiUrl: HasTestEnv + Send + Sync {
 impl HasPublicApiUrl for IcNodeSnapshot {
     fn get_public_url(&self) -> Url {
         let node_record = self.raw_node_record();
-        IcNodeSnapshot::http_endpoint_to_url(&node_record.http.expect("Node doesn't have URL"))
+
+        // API boundary nodes listen on port 443, while replicas listen on port 8080
+        if self.is_api_boundary_node() {
+            Url::parse(&format!("https://[{}]", self.get_ip_addr())).expect("Failed to parse URL")
```

Comment:
[get_ip_addr](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/driver/src/driver/test_env_api.rs?L826:12-826:23) returns an IpAddr which could "type-wise" be an IPv4 address in which case the `[]` around the address are incorrect. But maybe this is safe because IC nodes can never have an IPv4 address?

Reply:
In case it's safe it would be nice to explicitly convert the `IpAddr` to `Ipv6Addr` so that we get a panic in case this invariant is broken.